klippy/kinematics/extruder.py

```diff
@@ -185,6 +191,29 @@ def cmd_SET_PRESSURE_ADVANCE(self, params):
                    pressure_advance, smooth_time))
         self.printer.set_rollover_info(self.name, "%s: %s" % (self.name, msg))
         gcode.respond_info(msg, log=False)
+    cmd_GET_E_STEP_DISTANCE_help = "Get extruder step distance"
+    def cmd_GET_E_STEP_DISTANCE(self, params):
+        gcode = self.printer.lookup_object('gcode')
+        step_dist = self.stepper.get_step_dist()
+        gcode.respond_info("%s E step distance: %f" % (self.name, step_dist))
+    cmd_SET_E_STEP_DISTANCE_help = "Set extruder step distance"
+    def cmd_SET_E_STEP_DISTANCE(self, params):
+        gcode = self.printer.lookup_object('gcode')
+        toolhead = self.printer.lookup_object('toolhead')
+        dist = gcode.get_float('DISTANCE', params, 0.)
+        old_step_dist = self.stepper.get_step_dist()
+        if abs((dist - old_step_dist)/old_step_dist) > 0.25:
+            gcode.respond_info("Failed: step change is outside of"
+                               " safe margin. %f -> %f" %
+                                            (old_step_dist, dist))
+        else:
```

Comment:
FWIW, I don't think this check is necessary.

Reply:
I agree, when I implemented that it was a knee jerk reaction to making a mistake typing in a value too high, but with some of the other uses this might provide the check would only be a hindrance.  I will remove it.